src/main/java/org/prebid/server/auction/BidResponseCreator.java

```diff
@@ -154,6 +155,17 @@ public BidResponseCreator(CacheService cacheService,
                                boolean debugEnabled) {
 
         final long auctionTimestamp = auctionTimestamp(auctionContext);
+        final Account account = auctionContext.getAccount();
+
+        final EventsContext eventsContext = EventsContext.builder()
+                .enabledForAccount(eventsEnabledForAccount(auctionContext))
+                .enabledForRequest(eventsEnabledForRequest(auctionContext))
+                .auctionTimestamp(auctionTimestamp)
+                .integration(integrationFrom(auctionContext))
+                .build();
+
+        final Map<String, String> bidIdToGeneratedBidId = new HashMap<>();
+        updateBidAdmInBidderResponses(bidderResponses, account, bidIdToGeneratedBidId, eventsContext);
```

Comment:
Should all these actions follow after `isEmptyBidderResponses(..)` method call?

Reply:
It is was intentional to include hook execution after this update and allow it to create new bidderResponses if needed